---
title: "99_biological insights"
format: html
editor: visual
---

# Dataset1

## DESeq2

### FSC

All of these pathways are related to **ribosome biogenesis and RNA metabolism/processing**, including:

-   **rRNA and tRNA metabolism/processing** (`GOBP_RRNA_METABOLIC_PROCESS`, `GOBP_TRNA_METABOLIC_PROCESS`, `GOBP_MATURATION_OF_SSU_RRNA`, `GOBP_TRNA_PROCESSING`)

-   **Ribosomal subunit biogenesis** (`GOBP_RIBOSOMAL_LARGE_SUBUNIT_BIOGENESIS`, `GOBP_RIBOSOMAL_SMALL_SUBUNIT_BIOGENESIS`)

-   **Ribonucleoprotein complex assembly** (`GOBP_RIBONUCLEOPROTEIN_COMPLEX_SUBUNIT_ORGANIZATION`)

-   **ncRNA processing and RNA localization** (`GOBP_NCRNA_PROCESSING`, `GOBP_RNA_LOCALIZATION`)

    The top GSEA hits suggest that the treatment primarily affects **ribosome biogenesis and RNA processing (translation and RNA metabolism) programs**.

## DEXSeq

## DE v.s DIU

### Gene level

Among the 12,089 genes shared between the two analyses, the overlap of significant genes is strongly enriched. Fisherâ€™s exact test yields a p-value \< 2.2e-16 and an odds ratio of 1.65, indicating that genes significant in both DESeq2 and DEXSeq occur far more frequently than expected by chance. Notably, 710 genes are significant in DEXSeq but not in DESeq2, highlighting how isoform-level analysis can reveal regulatory changes in transcript usage that may be missed by gene-level differential expression.

### Gene-set level

GSEA results show *no significant pathways* when using DEXSeq gene-level statistics.
